neon/datasets/cifar100.py

```diff
@@ -0,0 +1,126 @@
+# ----------------------------------------------------------------------------
+# Copyright 2014 Nervana Systems Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------
+"""
+CIFAR-100 contains color images of 10 classes.
+More info at: http://www.cs.toronto.edu/~kriz/cifar.html
+"""
+
+import logging
+import numpy as np
+import os
+import tarfile
+
+from neon.datasets.dataset import Dataset
+from neon.util.compat import range
+from neon.util.persist import deserialize
+
+
+logger = logging.getLogger(__name__)
+
+
+class CIFAR100(Dataset):
+
+    """
+    Sets up a CIFAR-100 dataset.
+
+    Attributes:
+        url (str): where to find the source data
+        backend (neon.backends.Backend): backend used for this data
+        inputs (dict): structure housing the loaded train/test/validation
+                       input data
+        targets (dict): structure housing the loaded train/test/validation
+                        target data
+
+    Kwargs:
+        repo_path (str, optional): where to locally host this dataset on disk
+    """
+    url = 'http://www.cs.utoronto.ca/~kriz/cifar-100-python.tar.gz'
+
+    def __init__(self, coarse=False, **kwargs):
+        self.macro_batched = False
+        self.coarse = coarse
```

Comment:
Can you describe the `coarse` parameter in the docstring Attributes section (perhaps renaming to something more descriptive like `coarse_targets`?)  Also, instead of having it be part of the named constructor arguments, just let it be picked up and updated like the other kwargs.  The following should work:

    def __init__(self, **kwargs):
        self.macro_batched = False
        self.coarse = False
        self.__dict__.update(kwargs)